### PR TITLE
Configmap metadataprovider

### PR DESCRIPTION
**Pull request by nahguam**
Stacked on #135 

True diff: https://github.com/streamnative/oxia/compare/bind-address...configmap-metadataprovider

### CODE REVIEW

File: coordinator/impl/metadata_configmap.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	}
+
+	if version != expectedVersion {
+		err = ErrUnexpectedVersion
+		return
+	}
+
+	version++
+
+	err = kubernetes.ConfigMaps(m.kubernetes).Upsert(m.namespace, configMap(m.name, status, version))
```
Comment (merlimat) [inline coordinator/impl/metadata_configmap.go]: If the version check is not supported by the ConfigMap API, we would have a race condition here if there are 2 coordinator processes running at the same time, and we would lose some of the updates

Comment (nahguam) [reply coordinator/impl/metadata_configmap.go]: Good point. I believe there might be a revision property that can be checked.

Comment (nahguam) [reply coordinator/impl/metadata_configmap.go]: There isn't on cm resources. But even if there were, I don't see a way to atomically update the resource. The same is true on CRD status. Actually not sure whether the `resourceVersion` applies there. Will need to read up a bit more

Comment (nahguam) [reply coordinator/impl/metadata_configmap.go]: @merlimat Yes we can use the `resourceVersion`, as long as it's treated as opaque. It currently maps to the etcd equivalent but they might have plans to change that in the future. It's actually represented as a `string` in the model, but parsed to a `uint` internally. 

I'll change the data type to a `string` to match the k8s go-client API:

`type Version string`

Comment (merlimat) [reply coordinator/impl/metadata_configmap.go]: 👍

**Pull request merged**